app/controllers/api/v1/pages_controller.rb

```diff
@@ -39,8 +39,11 @@ def require_admin
       end
 
       def permitted_params
+        if params[:social_image].present? && params[:social_image][:url].present?
```

Comment:
non-blocking nit: there's a nice `dig` method for this.
```suggestion
        if params.dig(:social_image, :url).present?
```